job_controller/job.go

```diff
@@ -93,14 +93,14 @@ func (jc *JobController) ReconcileJobs(
 
 	oldStatus := jobStatus.DeepCopy()
 
-	pods, err := jc.GetPodsForJob(metaObject)
+	pods, err := jc.Controller.GetPodsForJob(job)
```

Comment:
Do we need to specify `Controller` here? Interface embedding will do it for us, I think.

Reply:
that's due to the [controller field](https://github.com/kubeflow/common/blob/c99eb21e464ea0365b77a1172f5e7e8c4468cb11/job_controller/job_controller.go#L117) is explicitly used in the struct. 

this was there originally, we can remove it in a separate PR.

Reply:
Gotcha. SGTM